happybase/client.py

```diff
@@ -28,41 +43,70 @@ def _req(self, _api, *args, **kwargs):
         interval = 0
         client = super(RecoveringClient, self)
         while True:
-            t = time()
+            t = utc_timestamp()
             try:
                 return client._req(_api, *args, **kwargs)
             except (TApplicationException, socket_error, TTransportException) as exc:
                 logger.exception("Got exception")
                 while True:
                     interval = retries.popleft() if retries else interval
                     logger.info("Sleeping for %d seconds", interval)
-                    sleep(interval)
-                    logger.info("Trying to reconnect")
-                    try:
-                        self._connection._refresh_thrift_client()
-                        self._connection.open()
-                        client = super(RecoveringClient, self._connection.client)
-                        logger.debug("New client is initialized")
-                    except TTransportException:
-                        logger.exception("Got exception, while trying to reconnect. Continuing")
-                        pass
-                    else:
+                    time.sleep(interval)
+                    client = self._reopen_connection()
+                    if client:
                         break
                 if no_retry:
                     raise exc
             finally:
-                now = time()
-                elapsed = (now - t) * 1E+3
+                now = utc_timestamp()
+                elapsed = int((now - t) * 1E+3)
```

Comment:
Not really, you're right, we do need floats for timestamps.
Don't think we need floats for elapsed though, it's already in milliseconds.